StuddyBuddy.py

```diff
@@ -0,0 +1,162 @@
+            self.sayAnimated(schedule)
+            self.textLock.acquire()
+            # leaving out the "sending email" and the "reschedule" part.
+
+            # End conversation with motivational quote
+            self.tellRandomMotivationQuote()
+
+        # Student seems to be doing fine (not anxious). No scheduling needed
+        else:
+            # TODO: Configure yes_no intend in DialogFlow
+            self.ask('It seems like you are quite positive today. Do you still need any motivation?', 'yes_no')
+            if self.yesAnswer:
+                self.tellRandomMotivationQuote()
+
+        self.stop()
+
+    def onAudioIntent(self, *args, intentName):
+        if len(args) > 0:
+            self.intendUnderstood = True
+            if intentName == 'activation':
+                self.activation = True
+            elif intentName == 'students_feeling':
+                self.studentsFeeling = list(args)
+            elif intentName == 'yes_no':
+                if args[0] == 'yes':
+                    self.yesAnswer = True
+                else:
+                    self.yesAnswer = False
+            # elif intentName == 'changing_wish':
+            #     pass
+            # elif intentName == 'schedule':
+            #     pass
+            elif intentName == 'time_left':
+                self.timeLeft = args[0]
+            elif intentName == 'to_do':
+                self.toDos = list(args)
+
+    def onRobotEvent(self, event):
+        # make sure all our started actions are completed:
+        if event == 'TextDone':
+            self.textLock.release()
+        elif event == 'LanguageChanged':
+            self.languageLock.release()
+
+    def ask(self, question, audioContext, attempts=3, timeout=5):
+        # We only want the question to be asked once, right?
+        self.sayAnimated(question)
+        self.textLock.acquire()
+
+        # new question, new stuff to understand
+        self.intendUnderstood = False
+        while attempts > 0 and not self.intendUnderstood:
+            attempts -= 1
+            self.setAudioContext(audioContext)
+            self.startListening()
+            self.intentLock.acquire(timeout=timeout)
+            self.stopListening()
+            if not self.intendUnderstood and attempts > 0:
+                self.sayAnimated('Sorry, I didn\'t catch that. Could you please repeat that?')
+                self.textLock.acquire()
+        if attempts == 0:
+            self.sayAnimated('Sorry. There seems to be a problem. I am shutting down.')
+            raise InteractionException
+
+    def studentIsAnxious(self):
+        # TODO: implement this
+        # Here, we want to do the sentiment analysis
+        raise NotImplementedError
+
+    def tellRandomMotivationQuote(self):
+        quotes = ['The will to win, the desire to succeed, the urge to reach your full potential.Tthese are the keys that will unlock the door to personal excellence.',
```

Comment:
Typo

Reply:
Noticed the above typo. Also, the functions follow camelCase, which is non-pythonic, but consistent with their API. I feel very dirty doing this, but I think consistency trumps aesthetics in this case. 

Otherwise this looks like a good start. Once we've got this working, we can flesh things out more.

Reply:
Nice work, @sa-and 💪